json_spec/README.md

```diff
@@ -1,16 +1,2589 @@
+        <td>Validates that a <a href="#measurement"><em>measurement</em></a> and a comparison value are not equal.</td>
+    </tr>
+    <tr>
+        <td><a name="validatortype-lessthan"><strong>LESS_THAN</strong></a></td>
+        <td>Validates that a <a href="#measurement"><em>measurement</em></a> is less than a comparison value.</td>
+    </tr>
+    <tr>
+        <td><strong>LESS_THAN_OR_EQUAL</strong></td>
+        <td>Validates that a <a href="#measurement"><em>measurement</em></a> is less than or equal to a comparison value.</td>
+    </tr>
+    <tr>
+        <td><strong>GREATER_THAN</strong></td>
+        <td>Validates that a <a href="#measurement"><em>measurement</em></a> is greater than a comparison value.</td>
+    </tr>
+    <tr>
+        <td><strong>GREATER_THAN_OR_EQUAL</strong></td>
+        <td>Validates that <a href="#measurement"><em>measurement</em></a> is greater than or equal to a comparison value.</td>
+    </tr>
+    <tr>
+        <td><strong>REGEX_MATCH</strong></td>
+        <td>Validates that a <a href="#measurement"><em>measurement</em></a> is found to match one or more regex strings contained by the <a href="#validator"><em>validator</em></a>.</td>
+    </tr>
+    <tr>
+        <td><strong>REGEX_NO_MATCH</strong></td>
+        <td>Validates that a <a href="#measurement"><em>measurement</em></a> does not match any regex strings contained by the <a href="#validator"><em>validator</em></a>.</td>
+    </tr>
+    <tr>
+        <td><strong>IN_SET</strong></td>
+        <td>Validates that a <a href="#measurement"><em>measurement</em></a> exists within a set of allowable values contained by the <a href="#validator"><em>validator</em></a>.</td>
+    </tr>
+    <tr>
+        <td><strong>NOT_IN_SET</strong></td>
+        <td>Validates that a <a href="#measurement"><em>measurement</em></a> does not exist within a set of allowables values contained by the <a href="#validator"><em>validator</em></a>.</td>
+    </tr>
+</table>
+
+
+### Identifiers
+
+Due to the streaming nature of the [**specification**](#def-spec), a common implementation pattern is the adoption of an identifier to indicate a relationship between an emitted artifact and a feature of the [**system under test**](#def-dut). Examples include identifiers for hardware components and software elements of the [**device under test**](#def-dut).
+
+In both cases, these hardware/software related identifiers are emitted at the beginning of the test output so that they can be referenced from subsequent artifacts.
+
+In addition, each [_Test Step_](#test-step-artifacts), [_measurement_](#measurement) and time series [_measurement element_](#measurementserieselement) also have an id attribute to allow other [**artifacts**](#def-artifact) to reference them by association, such as an [_error_](#error) being associated to a particular software element, or a [_diagnosis_](#diagnosis) being associated to one or more [hardware components](#hardwareinfo).
```

Comment:
nit: punctuation is different for Test Step vs. the other items which are lower case.

Reply:
done. please check/resolve